chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/impl/PendingSurveillanceManagerImpl.java

```diff
@@ -0,0 +1,728 @@
+        PendingSurveillanceEntity pendingSurv = survDao.getPendingSurveillanceById(pendingSurvId, true);
+        return isPendingSurveillanceAvailableForUpdate(pendingSurv);
+    }
+
+    private boolean isPendingSurveillanceAvailableForUpdate(final PendingSurveillanceEntity pendingSurv)
+            throws EntityRetrievalException, ObjectMissingValidationException {
+        if (pendingSurv.getDeleted()) {
+            ObjectMissingValidationException alreadyDeletedEx = new ObjectMissingValidationException();
+            alreadyDeletedEx.getErrorMessages()
+            .add("This pending surveillance has already been confirmed or rejected by another user.");
+            alreadyDeletedEx.setObjectId(pendingSurv.getId().toString());
+            alreadyDeletedEx.setStartDate(pendingSurv.getStartDate());
+            alreadyDeletedEx.setEndDate(pendingSurv.getEndDate());
+
+            try {
+                UserDTO lastModifiedUser = userDao.getById(pendingSurv.getLastModifiedUser());
+                if (lastModifiedUser != null) {
+                    Contact contact = new Contact();
+                    contact.setFullName(lastModifiedUser.getFullName());
+                    contact.setFriendlyName(lastModifiedUser.getFriendlyName());
+                    contact.setEmail(lastModifiedUser.getEmail());
+                    contact.setPhoneNumber(lastModifiedUser.getPhoneNumber());
+                    contact.setTitle(lastModifiedUser.getTitle());
+                    alreadyDeletedEx.setContact(contact);
+                } else {
+                    alreadyDeletedEx.setContact(null);
+                }
+            } catch (final UserRetrievalException ex) {
+                alreadyDeletedEx.setContact(null);
+            }
+            throw alreadyDeletedEx;
+        }
+        //If pendingSurv were null, we would have gotten an NPE by this point
+        //return pendingSurv != null;
+        return true;
+    }
+
+    private Integer getSurveillanceRecordThreshold() {
+        String surveillanceThresholdToProcessAsJobStr = env.getProperty("surveillanceThresholdToProcessAsJob").trim();
+        Integer surveillanceThresholdToProcessAsJob = SURV_THRESHOLD_DEFAULT;
+        try {
+            surveillanceThresholdToProcessAsJob = Integer.parseInt(surveillanceThresholdToProcessAsJobStr);
+        } catch (final NumberFormatException ex) {
+            LOGGER.error(
+                    "Could not format " + surveillanceThresholdToProcessAsJobStr + " as an integer. Defaulting to"
+                            + " 50 instead.");
+        }
+        return surveillanceThresholdToProcessAsJob;
+    }
+
+    private void validate(final Surveillance surveillance) {
+        validator.validate(surveillance, false);
+    }
+
+    private void checkSurveillanceAuthority(final Surveillance surv) throws SurveillanceAuthorityAccessDeniedException {
+        Boolean hasOncAdmin = Util.isUserRoleAdmin();
```

Comment:
Should this be Util.isUserRoleOnc? Or should the boolean be named hasAdmin?

Reply:
Addressed.